core/src/main/java/org/elasticsearch/action/admin/indices/forcemerge/ForceMergeRequest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.forcemerge;
+
+import org.elasticsearch.action.support.broadcast.BroadcastRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * A request to force merging of one or more indices. In order to run amerge on
+ * all the indices, pass an empty array or <tt>null</tt> for the indices.
+ * <p/>
+ * <p>{@link #maxNumSegments(int)} allows to control the number of segments to
+ * force merge down to. By default, will cause the merge process to merge
+ * down to half the configured number of segments.
```

Comment:
This could use some rewording. I know it probably came from a copy but it could use mentioning that its per shard. Also, the half thing is new to me.